frontend/src/lib/components/AddInsightsToDashboard/AddInsightsToDashboard.tsx

```diff
@@ -0,0 +1,28 @@
+import { LemonButton } from '@posthog/lemon-ui'
+
+import { DashboardType } from '~/types'
+
+interface AddInsightsToDashboardProps {
+    setAddInsightsToDashboardModalOpen: (open: boolean) => void
+    dashboard: DashboardType
+    disabledReason: string | null
+}
+
+export function AddInsightsToDashboard({
+    setAddInsightsToDashboardModalOpen,
+    dashboard,
+    disabledReason,
+}: AddInsightsToDashboardProps): JSX.Element | null {
+    return (
+        <LemonButton
+            onClick={() => {
+                setAddInsightsToDashboardModalOpen(true)
+            }}
+            type="primary"
+            data-attr="insight-add-graph"
+            disabledReason={disabledReason}
+        >
+            {dashboard.tiles.every((tile) => !tile.insight) ? 'Add Insight' : 'Manage Insights'}
```

Comment:
Since insights can be removed from the dashboard on the dashboard itself, I think it'd be best to keep this as "Add insight". Please note that we also don't capitalize "insights" or "dashboards".